lib/dns.js

```diff
@@ -309,21 +309,23 @@ function setServers(servers) {
   // servers cares won't have any servers available for resolution
   const orig = cares.getServers();
   const newSet = [];
+  const IPv6RE = /\[(.*)\]/;
+  const addrSplitRE = /:\d+$/;
 
   servers.forEach((serv) => {
     var ipVersion = isIP(serv);
     if (ipVersion !== 0)
       return newSet.push([ipVersion, serv]);
 
-    const match = serv.match(/\[(.*)\](?::\d+)?/);
```

Comment:
Maybe this was for some more clarity, to show the wider pattern. But this may impact performance a bit.